### PR TITLE
purge: Remove all gpg files during apt purge (SC-537)

### PR DESCRIPTION
**Pull request by CalvoM**
## Proposed Commit Message

purge: remove all the gpg files during apt purge

Now we can delete the gpg files in **/etc/apt/trusted.gpg.d/** folder belonging to ubuntu-advantage-tools.

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: debian/ubuntu-advantage-tools.postrm [hunk 1]
```diff
@@ -19,9 +19,7 @@ remove_logs(){
 }
 
 remove_gpg_files(){
-    rm -f /etc/apt/trusted.gpg.d/ubuntu-advantage-esm-infra-trusty.gpg
-    rm -f /etc/apt/trusted.gpg.d/ubuntu-advantage-esm-apps.gpg
-    rm -f /etc/apt/trusted.gpg.d/ubuntu-advantage-fips.gpg
+    rm -f /etc/apt/trusted.gpg.d/ubuntu-advantage-*.gpg
```
Comment (renanrodrigo) [inline debian/ubuntu-advantage-tools.postrm:22]: The only problem here is removing files we should not. Given that, I think that no other APT repository should have `ubuntu-advantage` as a prefix for the keys, and that it is safe and reasonable for us to remove it like this.

**Review by renanrodrigo — APPROVED** [review]
Thanks, @CalvoM! LGTM.
Please just remove the unrelated commit from this branch, and we are good.

File: debian/ubuntu-advantage-tools.postrm (hunk 1 above)
Comment (lucasmoura) [reply debian/ubuntu-advantage-tools.postrm:22]: I agree with that approach, but let's just confirm with @blackboxsw to see if this is really safe

**Review by blackboxsw — APPROVED** [review]
I agree here. Our GPG keys are scoped with the "ubuntu-advantage-" prefix and other software shouldn't be using this prefix as they risk being overwritten by our own utilities. +1

**Comment by renanrodrigo** [comment]
CI failure unrelated - landing this

**Pull request merged**